e2e/tests/functional/couchdb.e2e.spec.js

```diff
@@ -71,3 +71,38 @@ test.describe("CouchDB Status Indicator @couchdb", () => {
     });
 });
 
+test.describe("CouchDB initialization @couchdb", () => {
+    test.use({ failOnConsoleError: false });
+    test("'My Items' folder is created if it doesn't exist", async ({ page }) => {
+        // Store any relevant PUT requests that happen on the page
+        const createMineFolderRequests = [];
+        page.on('request', req => {
+            // eslint-disable-next-line playwright/no-conditional-in-test
+            if (req.method() === 'PUT' && req.url().endsWith('openmct/mine')) {
+                createMineFolderRequests.push(req);
+            }
+        });
+
+        // Override the first request to GET openmct/mine to return a 404
+        await page.route('**/openmct/mine', route => {
+            route.fulfill({
+                status: 404,
+                contentType: 'application/json',
+                body: JSON.stringify({})
+            });
+        }, { times: 1 });
+
+        // Go to baseURL
+        await page.goto('./', { waitUntil: 'networkidle' });
+
+        // Verify that error banner is displayed
+        const bannerMessage = await page.locator('.c-message-banner__message').innerText();
+        expect(bannerMessage).toEqual('Failed to retrieve object mine');
+
+        // Verify that a PUT request to create "My Items" folder was made
+        expect.poll(() => createMineFolderRequests.length, {
```

Comment:
Don't know how playwright works, but just wondering whether this can be achieved with a Promise resolved by the logic on line 81 rather than polling? It might be a bit cleaner than polling?

Reply:
So at this point in the test, I'm trying to verify that Open MCT made an attempt to create the 'mine' folder. My other option was to use `page.waitForRequest()` (https://playwright.dev/docs/api/class-page#page-wait-for-request) with the same specified criteria.

The problem with that approach is that if that request is never made, all we get in the test failure is a generic timeout message. Using `.poll()` solves two problems here:

1. Using `.poll()` will stabilize the test. Sometimes the test happens so fast that the page event isn't captured by the time that the assertion resolves. The `.poll()` has a very short timeout of 1sec so it will retry the assertion if it fails at first (and is still within the timeout).
2. We can specify a message to make it more clear what exactly failed here.

Reply:
So, in Jasmine I would do this like so -

``` JavaScript
test.describe("CouchDB initialization @couchdb", () => {
    test.use({ failOnConsoleError: false });
  let resolveMyItems;
  
  const awaitCreationOfMyItems = new Promise((resolve, reject) => {
      resolveMyItems = resolve;
  });
  //...
  page.on('request', req => {
      // eslint-disable-next-line playwright/no-conditional-in-test
      if (req.method() === 'PUT' && req.url().endsWith('openmct/mine')) {
          resolveMyItems();
      }
  });
  //...
  
  return awaitCreationOfMyItems;
}
```

Jasmine used to use polling for asynchronous events as well, but moved to an event driven model. I don't know if Playwright supports returning Promises in the same way that Jasmine does though?

Reply:
Will merge for now, but would like to explore this a little further and establish a consistent pattern.